docs/contributing/999-rfcs/2022-12-08-wing-inputs.md

```diff
@@ -0,0 +1,202 @@
+	app.addInput("S3BackendBucket", {name: `${rootKey}/bucket`});
+	app.addInput("S3BackendStateFile", {name: `${rootKey}/state-file`});
+  }
+}
+```
+
+Compiling the following platform with missing inputs will result in an error message like such:
+```
+Missing required inputs:
+/tf-s3-backend/bucket
+/tf-s3-backend/state-file-name
+```
+
+To satisfy the required inputs we would need to create a `wing.toml` with those required values
+```toml
+[ tf-s3-backend ]
+bucket = "my-wing-bucket"
+state-file = "my state file"
+```
+
+### Creating inputs in preflight
+
+Sometimes we may just want to provide an input defined within our Wing application. Take for instance the following 
+```js
+let testEmail = new cloud.Input(name: "/test/email", defaultValue: "test@fake.com");
+
+new cloud.Function(inflight() => {
+	// some code that uses the testEmail
+});
+```
+
+While more adhoc than creating an `wing.toml` file, this should be fine in most non-secure cases.
+### Secure Inputs
+
+Some inputs contain sensitive data and thus should be treated with enhanced scrutiny. I.E. Inputs with sensitive data should not be created by preflight code in order to keep any trace of them from state files. 
+
+A secure input can be indicated using the optional prop, `secure` as such
+```js
+new cloud.Input({name: "/my-db/connection-string", secure: true})
+```
+
+This will result in an error like 
+```
+Cannot create secure Inputs in Preflight
+... followed by some Platform specific text on how secure inputs are expected to be created
+```
+
+### Creating the inputs Interactively (P2)
+
+Additionally inputs can be created through an interactive process by running `wing compile -i app.main.w` 
+
+This interactive experience will prompt users to provide values for all of their inputs
+```sh
+wing compile -i app.main.w
+
+Please enter an input value for /features/notifications: "true"
+Please enter an input value for /email-list: "one@email.com,two@email.com"
+```
+
+As well re-running this command will give the opportunity to change input values
+```sh
+wing compile -i app.main.w
+
+Please enter an input value for /features/notifications (true):
+Please enter an input value for /email-list (one@email.com,two@email.com): ""
+```
+By default, not entering in a new value will keep the existing version.
+
+### List Application Inputs
+
+Its also possible to view a list of the inputs for a given application and target platforms. This can be achieved by using the `wing inputs --list` command 
+
+For example using the below application code
+
+```js
+let enableNotifications = cloud.Input("/features/notifications");
+let emailList = cloud.Input("/email-list"); // comma seperated string
+```
+
+combined with a Platform that has required inputs (such as the TerraformS3Backend one mentioned above) can be done with 
+```bash
+wing inputs --list -t tf-aws -t @tf-s3-backend
+```
+
+will produce an output like this:
+```
+Application inputs:
+/features/notifications
```

Comment:
I would recommend adding a section which describes how inputs are implemented in various platforms (at least `sim` and `aws`).